src/main/java/ch/njol/skript/effects/EffChange.java

```diff
@@ -100,178 +99,180 @@ public class EffChange extends Effect {
+				if (changeMode == ChangeMode.SET)
+					Skript.error(expressionToChange + " can only be set to one " + Classes.getSuperClassInfo(superTypeOfAcceptableChangeTypes).getName() + ", not more", ErrorQuality.SEMANTIC_ERROR);
 				else
-					Skript.error("only one " + Classes.getSuperClassInfo(x).getName() + " can be " + (mode == ChangeMode.ADD ? "added to" : "removed from") + " " + changed + ", not more", ErrorQuality.SEMANTIC_ERROR);
+					Skript.error("Only one " + Classes.getSuperClassInfo(superTypeOfAcceptableChangeTypes).getName() + " can be " + (changeMode == ChangeMode.ADD ? "added to" : "removed from") + " " + expressionToChange + ", not more", ErrorQuality.SEMANTIC_ERROR);
 				return false;
 			}
 
-			if (changed instanceof Variable && !changed.isSingle() && mode == ChangeMode.SET) {
-				if (ch instanceof ExprParse) {
-					((ExprParse) ch).flatten = false;
-				} else if (ch instanceof ExpressionList) {
-					for (Expression<?> expression : ((ExpressionList<?>) ch).getExpressions()) {
+			if (expressionToChange instanceof Variable && !expressionToChange.isSingle() && changeMode == ChangeMode.SET) {
+				if (deltaValuesExpression instanceof ExprParse) {
+					((ExprParse) deltaValuesExpression).flatten = false;
+				} else if (deltaValuesExpression instanceof ExpressionList) {
+					for (Expression<?> expression : ((ExpressionList<?>) deltaValuesExpression).getExpressions()) {
 						if (expression instanceof ExprParse)
 							((ExprParse) expression).flatten = false;
 					}
 				}
 			}
 
-			if (changed instanceof Variable && !((Variable<?>) changed).isLocal() && (mode == ChangeMode.SET || ((Variable<?>) changed).isList() && mode == ChangeMode.ADD)) {
-				final ClassInfo<?> ci = Classes.getSuperClassInfo(ch.getReturnType());
-				if (ci.getC() != Object.class && ci.getSerializer() == null && ci.getSerializeAs() == null && !SkriptConfig.disableObjectCannotBeSavedWarnings.value()) {
+			if (expressionToChange instanceof Variable && !((Variable<?>) expressionToChange).isLocal() && (changeMode == ChangeMode.SET || ((Variable<?>) expressionToChange).isList() && changeMode == ChangeMode.ADD)) {
+				final ClassInfo<?> deltaReturnTypeClassInfo = Classes.getSuperClassInfo(deltaValuesExpression.getReturnType());
+				if (deltaReturnTypeClassInfo.getC() != Object.class && deltaReturnTypeClassInfo.getSerializer() == null && deltaReturnTypeClassInfo.getSerializeAs() == null && !SkriptConfig.disableObjectCannotBeSavedWarnings.value()) {
 					if (getParser().isActive() && !getParser().getCurrentScript().suppressesWarning(ScriptWarning.VARIABLE_SAVE)) {
-						Skript.warning(ci.getName().withIndefiniteArticle() + " cannot be saved, i.e. the contents of the variable " + changed + " will be lost when the server stops.");
+						Skript.warning(deltaReturnTypeClassInfo.getName().withIndefiniteArticle() + " cannot be saved, i.e. the contents of the variable " + expressionToChange + " will be lost when the server stops.");
 					}
 				}
```

Comment:
Maybe it's just github UI messing up but it looks like we have two ifs nested in here and neither of them have an `else`, maybe they could be combined or tidied up somehow?